src/lib/pubkey/kyber/kyber_common/kyber.h

```diff
@@ -155,6 +173,17 @@ class BOTAN_PUBLIC_API(3, 0) Kyber_PrivateKey final : public virtual Kyber_Publi
       std::unique_ptr<PK_Ops::KEM_Decryption> create_kem_decryption_op(RandomNumberGenerator& rng,
                                                                        std::string_view params,
                                                                        std::string_view provider) const override;
+      /**
+       * Get the private key bytes in expanded format as specified in FIPS 203.
+       */
+      secure_vector<uint8_t> expanded_private_key_bits() const;
+
+      /**
+       * Get the private key bytes as a 64-byte seed: d || z. If the private key
+       * is a Kyber Round 3 key or was loaded using expanded key bytes
+       * std::nullopt is returned.
+       */
+      std::optional<secure_vector<uint8_t>> seed_private_key_bits() const;
```

Comment:
Possible alternate interface here

```
enum class PrivateKeyFormat {
  Seed,
  Expanded,
};

// throws Encoding_Error if not available
secure_vector<uint8_t> private_key_bits_with_format(PrivateKeyFormat format) const;
```

Something similar could be done in any other cases where the key has multiple formats.

Reply:
Fair. Especially since we will likely also need that for ML-DSA. And might also be useful for the PKCS#8 factory approach.